src/docfx/build/document/DocumentProvider.cs

```diff
@@ -138,8 +107,7 @@ private bool TryGetDocumentIdConfig(PathString path, out DocumentIdConfig result
 
         private Document GetDocumentCore(FilePath path)
         {
-            var contentType = path.Origin == FileOrigin.Redirection ? ContentType.Redirection : GetContentType(path.Path);
-
+            var contentType = _buildScope.GetContentType(path);
```

Comment:
>GetContentType [](start = 42, length = 14)

minor suggestion: 
Can this be a static helper function and stay in DocumentProvider?(feels more reasonable to me)
`public static GetContentType(FilePath path)`